Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -10,243 +10,424 @@
+      rd_info->setResidueName(m_residue_name->at(atom_num));
+    }
 
-    shared_ptr<IndexedStringProperty> m_atom_name;
-    shared_ptr<IndexedStringProperty> m_residue_name;
-    shared_ptr<IndexedStringProperty> m_chain_id;
-    shared_ptr<IndexedStringProperty> m_insertion_code;
+    if (m_chain_id && m_chain_id->isDefined(atom_num)) {
+      rd_info->setChainId(m_chain_id->at(atom_num));
+    }
 
-    shared_ptr<IndexedIntProperty> m_resnum;
+    if (m_insertion_code && m_insertion_code->isDefined(atom_num)) {
+      rd_info->setInsertionCode(m_insertion_code->at(atom_num));
+    }
 
-    shared_ptr<IndexedRealProperty> m_occupancy;
-    shared_ptr<IndexedRealProperty> m_tempfac;
+    if (m_resnum && m_resnum->isDefined(atom_num)) {
+      rd_info->setResidueNumber(m_resnum->at(atom_num));
+    }
 
-};
+    if (m_occupancy && m_occupancy->isDefined(atom_num)) {
+      rd_info->setOccupancy(m_occupancy->at(atom_num));
+    }
 
-PDBInfo::PDBInfo(const shared_ptr<const IndexedBlock>& atom_data)
-{
-    try {
-    m_atom_name = atom_data->getStringProperty("s_m_pdb_atom_name");
-    } catch (std::out_of_range &) { }
+    if (m_tempfac && m_tempfac->isDefined(atom_num)) {
+      rd_info->setTempFactor(m_tempfac->at(atom_num));
+    }
+  }
 
-    try {
-    m_residue_name = atom_data->getStringProperty("s_m_pdb_residue_name");
-    } catch (std::out_of_range &) { }
+ private:
+  std::shared_ptr<mae::IndexedStringProperty> m_atom_name;
+  std::shared_ptr<mae::IndexedStringProperty> m_residue_name;
+  std::shared_ptr<mae::IndexedStringProperty> m_chain_id;
+  std::shared_ptr<mae::IndexedStringProperty> m_insertion_code;
 
-    try {
-    m_chain_id = atom_data->getStringProperty("s_m_chain_name");
-    } catch (std::out_of_range &) { }
+  std::shared_ptr<mae::IndexedIntProperty> m_resnum;
 
-    try {
-    m_insertion_code = atom_data->getStringProperty("s_m_insertion_code");
-    } catch (std::out_of_range &) { }
+  std::shared_ptr<mae::IndexedRealProperty> m_occupancy;
+  std::shared_ptr<mae::IndexedRealProperty> m_tempfac;
+};
 
-    try {
-    m_resnum = atom_data->getIntProperty("i_m_residue_number");
-    } catch (std::out_of_range &) { }
+bool streamIsGoodOrExhausted(std::istream *stream) {
+  PRECONDITION(stream, "bad stream");
+  return stream->good() || (stream->eof() && stream->fail() && !stream->bad());
+}
 
-    try {
-    m_occupancy = atom_data->getRealProperty("r_m_pdb_occupancy");
-    } catch (std::out_of_range &) { }
+void parseChiralityLabel(RWMol &mol, const std::string &stereo_prop) {
+  boost::char_separator<char> sep{"_"};
+  boost::tokenizer<boost::char_separator<char>> tokenizer{stereo_prop, sep};
+
+  auto tItr = tokenizer.begin();
+
+  const int chiral_idx = FileParserUtils::toInt(*tItr) - 1;
+  Atom *chiral_atom = mol.getAtomWithIdx(chiral_idx);
+  CHECK_INVARIANT(chiral_atom != nullptr, "bad prop value");
+
+  unsigned nSwaps = 2;
+  switch (*((++tItr)->rbegin())) {
```

Comment:
Does the token have a `.back()` like `vector` and `string`?

Reply:
It is actually a string, so it does. I'll make this a bit clearer.